MediaBrowser.Providers/Trickplay/TrickplayImagesTask.cs

```diff
@@ -0,0 +1,109 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using MediaBrowser.Controller.Configuration;
+using MediaBrowser.Controller.Entities;
+using MediaBrowser.Controller.Library;
+using MediaBrowser.Controller.Trickplay;
+using MediaBrowser.Model.Entities;
+using MediaBrowser.Model.Globalization;
+using MediaBrowser.Model.Tasks;
+using Microsoft.Extensions.Logging;
+
+namespace MediaBrowser.Providers.Trickplay
+{
+    /// <summary>
+    /// Class TrickplayImagesTask.
+    /// </summary>
+    public class TrickplayImagesTask : IScheduledTask
+    {
+        private readonly ILogger<TrickplayImagesTask> _logger;
+        private readonly ILibraryManager _libraryManager;
+        private readonly ILocalizationManager _localization;
+        private readonly ITrickplayManager _trickplayManager;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="TrickplayImagesTask"/> class.
+        /// </summary>
+        /// <param name="logger">The logger.</param>
+        /// <param name="libraryManager">The library manager.</param>
+        /// <param name="localization">The localization manager.</param>
+        /// <param name="trickplayManager">The trickplay manager.</param>
+        public TrickplayImagesTask(
+            ILogger<TrickplayImagesTask> logger,
+            ILibraryManager libraryManager,
+            ILocalizationManager localization,
+            ITrickplayManager trickplayManager)
+        {
+            _libraryManager = libraryManager;
+            _logger = logger;
+            _localization = localization;
+            _trickplayManager = trickplayManager;
+        }
+
+        /// <inheritdoc />
+        public string Name => _localization.GetLocalizedString("TaskRefreshTrickplayImages");
+
+        /// <inheritdoc />
+        public string Description => _localization.GetLocalizedString("TaskRefreshTrickplayImagesDescription");
+
+        /// <inheritdoc />
+        public string Key => "RefreshTrickplayImages";
+
+        /// <inheritdoc />
+        public string Category => _localization.GetLocalizedString("TasksLibraryCategory");
+
+        /// <inheritdoc />
+        public IEnumerable<TaskTriggerInfo> GetDefaultTriggers()
+        {
+            return new[]
+            {
+                new TaskTriggerInfo
+                {
+                    Type = TaskTriggerInfo.TriggerDaily,
+                    TimeOfDayTicks = TimeSpan.FromHours(3).Ticks
+                }
+            };
+        }
+
+        /// <inheritdoc />
+        public async Task ExecuteAsync(IProgress<double> progress, CancellationToken cancellationToken)
+        {
+            var items = _libraryManager.GetItemList(new InternalItemsQuery
+            {
+                MediaTypes = new[] { MediaType.Video },
+                IsVirtualItem = false,
+                IsFolder = false,
+                Recursive = true
+            }).OfType<Video>().ToList();
+
```

Comment:
This should be paginated. No reason to pull all videos into memory at once.

Reply:
I don't entirely understand how pagination would work with this as I had assumed all videos/library data would be loaded and stay in memory server-side when the server starts, or at least as clients and other tasks start to request them.

For reference, this is the code from the chapter images task (https://github.com/jellyfin/jellyfin/blob/08aac575290d0b94c0c4b7e6af6167867dc93efa/Emby.Server.Implementations/ScheduledTasks/Tasks/ChapterImagesTask.cs#L90-L106) that I based the trickplay task off of. Do you have any examples of pagination for item queries as it relates to server-sided tasks that I could look at as a starting point?

Reply:
The subtitle extract plugin has pagination
https://github.com/jellyfin/jellyfin-plugin-subtitleextract/blob/master/Jellyfin.Plugin.SubtitleExtract/Tasks/ExtractSubtitlesTask.cs